merlin-framework-processor/src/main/java/gov/nasa/jpl/aerie/merlin/processor/generator/MapperMethodMaker.java

```diff
@@ -151,26 +153,58 @@ public final MethodSpec makeGetValidationFailuresMethod() {
             inputType.validations()
                 .stream()
                 .map(validation -> {
-                    final var subjects = Arrays.stream(validation.subjects())
-                        .map(subject -> CodeBlock.builder().add("$S", subject))
-                        .reduce((x, y) -> x.add(", ").add(y.build()))
-                        .orElse(CodeBlock.builder())
-                        .build();
+                    CodeBlock.Builder codeBlock = CodeBlock.builder().beginControlFlow("try");
 
-                    return CodeBlock
-                        .builder()
-                        .beginControlFlow("try")
-                        .addStatement(
-                            "if (!$L.$L()) notices.add(new $T($T.of($L), $S))",
-                            "input",
-                            validation.methodName(),
-                            ValidationNotice.class,
-                            List.class,
-                            subjects,
-                            validation.failureMessage())
-                        .nextControlFlow("catch ($T ex)", Throwable.class)
-                        .addStatement("notices.add(new $T($T.of($L), ex.getMessage()))", ValidationNotice.class, List.class, subjects)
-                        .endControlFlow();
+                    if (validation.isSimpleValidation()) {
+                        final var subjects = Arrays.stream(validation.subjects())
+                          .map(subject -> CodeBlock.builder().add("$S", subject))
+                          .reduce((x, y) -> x.add(", ").add(y.build()))
+                          .orElse(CodeBlock.builder())
+                          .build();
+
+                        codeBlock.addStatement(
+                          "if (!$L.$L()) notices.add(new $T($T.of($L), $S))",
+                          "input",
+                          validation.methodName(),
+                          ValidationNotice.class,
+                          List.class,
+                          subjects,
+                          validation.failureMessage());
+
+                        return codeBlock
+                          .nextControlFlow("catch ($T ex)", Throwable.class)
+                          .addStatement(
+                              "notices.add(new $T($T.of($L), ex.getMessage()))",
+                              ValidationNotice.class,
+                              List.class,
+                              subjects)
+                          .endControlFlow();
+                    }
+
+                    codeBlock.addStatement(
+                      "if (!$L.$L().$L()) notices.add(new $T($T.of($L.$L().$L()), $L.$L().$L().orElse(null)))",
```

Comment:
If `Optional.empty()` is going to converted to `null` here, why use `Optional` in `ValidationResult`?

Reply:
That's a good point, originally I wanted a `ValidationResult` to not have to include a `message`, but it's probably cleaner to just accept a `null` in that case.

`new ValidationResult(true, "biteSize", null)`

Reply:
Yeah, either an explicit `null` or adding an alternate constructor without a `message` field would work.

```java
public ValidationResult(boolean success, String subject) {
    this(success, subject, null);
  }
```